pkg/gardenlet/controller/backupentry/reconciler.go

```diff
@@ -599,19 +621,34 @@ func (r *Reconciler) newExtensionComponent(log logr.Logger, backupEntry *gardenc
 	)
 }
 
-func (r *Reconciler) waitUntilBackupBucketReconciled(ctx context.Context, log logr.Logger, backupBucket *gardencorev1beta1.BackupBucket) error {
-	return extensions.WaitUntilObjectReadyWithHealthFunction(
-		ctx,
-		r.GardenClient,
-		log,
-		health.CheckBackupBucket,
-		backupBucket,
-		"BackupBucket",
-		DefaultInterval,
-		DefaultSevereThreshold,
-		DefaultTimeout,
-		nil,
-	)
+func (r *Reconciler) checkIfBackupBucketIsHealthy(ctx context.Context, backupBucket *gardencorev1beta1.BackupBucket) error {
+	var lastObservedError error
+
+	if err := r.GardenClient.Get(ctx, client.ObjectKeyFromObject(backupBucket), backupBucket); err != nil {
+		return fmt.Errorf("failed getting associated BackupBucket %q: %w", backupBucket.Name, err)
+	}
+
+	if backupBucket.Status.LastOperation != nil {
+		// check for lastOperation and errors, and if none are present, BackupBucket is ready
+		lastOperationState := backupBucket.Status.LastOperation.State
+		if lastOperationState == gardencorev1beta1.LastOperationStateProcessing {
+			// the backupEntry will be requeued when the state of the BackupBucket changes to Succeeded
+			return nil
```

Comment:
Why does it return `nil` here? Shouldn't it return an error, so that the entire reconciliation ends?